### PR TITLE
new event `brewing starts`

### PR DESCRIPTION
**Pull request by pobab**
re: https://discord.com/channels/315163488085475337/1147019222346641449

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    public boolean applyDetermination(ScriptPath path, ObjectTag determinationObj) {
+        if (determinationObj instanceof DurationTag duration) {
+            event.setTotalBrewTime(duration.getTicksAsInt());
+            return true;
+        }
+        return super.applyDetermination(path, determinationObj);
+    }
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java]: Should use the modern `registerDetermination` methods (and probably have a prefix, e.g. `BREW_TIME:DurationTag`)

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    //
+    // @Location true
+    //
+    // @Triggers when a brewing stand starts brew a potion.
+    //
+    // @Context
+    // <context.location> returns a LocationTag of the brewing stand's location.
+    // <context.brew_time> returns a DurationTag of the total time it will take to brew the potion.
+    //
+    // @Determine to set the total time for the potion being brewed.
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java]: Need to follow proper determination meta format (can look at other events for examples)

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import com.denizenscript.denizencore.objects.core.DurationTag;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BrewingStartEvent;
+
+public class BrewingStartsScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // brewing stand starts brew
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java]: That event line sounds a bit weird imo, and also doesn't match what's in `registerCouldMatcher` - probably `brewing stand starts brewing`? or maybe `brewing starts`?

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    //
+    // @Group Block
+    //
+    // @Location true
+    //
+    // @Triggers when a brewing stand starts brew a potion.
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java]: `brewing`

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    //
+    // @Context
+    // <context.location> returns a LocationTag of the brewing stand's location.
+    // <context.brew_time> returns a DurationTag of the total time it will take to brew the potion.
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java]: The event seems to also expose a `getSource` `ItemStack` - what does that return? might be useful to add as a context?

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    //
+    // @Determine to set the total time for the potion being brewed.
+    //
+    // -->
+
+    public BrewingStartsScriptEvent(){
+        registerCouldMatcher("brewing starts");
+    }
+
+    public LocationTag location;
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java]: This field feels a bit redundant, can probably in-line it

File: plugin/src/main/java/com/denizenscript/denizen/events/ScriptEventRegistry.java [hunk 7]
```diff
@@ -76,6 +76,7 @@ public static void registerMainEvents() {
         ScriptEvent.registerScriptEvent(BlockShearEntityScriptEvent.class);
         ScriptEvent.registerScriptEvent(BlockSpreadsScriptEvent.class);
         ScriptEvent.registerScriptEvent(BrewingStandFueledScriptEvent.class);
+        ScriptEvent.registerScriptEvent(BrewingStartsScriptEvent.class);
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/events/ScriptEventRegistry.java]: It seems this event was only add in 1.19, so should have a version check here

File: plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    // <context.brew_time> returns a DurationTag of the total time it will take to brew the potion.
+    //
+    // @Determine
+    // "BREW_TIME:DurationTag" to set the total time for the potion being brewed.
+    //
+    // -->
+
+    public BrewingStartsScriptEvent() {
+        registerCouldMatcher("brewing starts");
+        this.<BrewingStartsScriptEvent, DurationTag>registerDetermination("brew_time", DurationTag.class, (evt, context, time) -> evt.event.setTotalBrewTime(time.getTicksAsInt()));
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/events/block/BrewingStartsScriptEvent.java:36]: should have `{}` around the block here

**Pull request merged**